server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy

```diff
@@ -1761,92 +1762,67 @@ class WorkflowController {
-                              sanitize( tipp.paymentType?.value ) + '\t' +  // access_type
-                              sanitize( tipp.title.getIdentifierValue('ZDB') ) +
-                              '\n');
+                writeExportLine(writer, sanitize, tipp, tipp)
               }
-              tipp.discard();
+              tipp.discard()
             }
           }
           tipps.close()
         }
-
-        writer.flush();
-        writer.close();
+        writer.flush()
+        writer.close()
       }
       out.close()
     }
     catch ( Exception e ) {
-      log.error("Problem with export",e);
+      log.error("Problem with export",e)
     }
   }
 
 
+  private writeExportLine(Writer writer, Closure<String> sanitize, TitleInstancePackagePlatform tipp, def tippCoverageStatement){
+    writer.write(
+        sanitize(tipp.title.name) + '\t' +
+            (tipp.title.hasProperty('dateFirstInPrint') ? sanitize(tipp.title.getIdentifierValue('pISBN')) : sanitize(tipp.title.getIdentifierValue('ISSN'))) + '\t' +
+            (tipp.title.hasProperty('dateFirstInPrint') ? sanitize(tipp.title.getIdentifierValue('ISBN')) : sanitize(tipp.title.getIdentifierValue('eISSN'))) + '\t' +
+            sanitize(tippCoverageStatement.startDate) + '\t' +
+            sanitize(tippCoverageStatement.startVolume) + '\t' +
+            sanitize(tippCoverageStatement.startIssue) + '\t' +
+            sanitize(tippCoverageStatement.endDate) + '\t' +
+            sanitize(tippCoverageStatement.endVolume) + '\t' +
+            sanitize(tippCoverageStatement.endIssue) + '\t' +
+            sanitize(tipp.url) + '\t' +
+            (tipp.title.hasProperty('firstAuthor') ? sanitize(tipp.title.firstAuthor) : '') + '\t' +
+            sanitize(tipp.title.getId()) + '\t' +
+            sanitize(tippCoverageStatement.embargo) + '\t' +
+            sanitize(tippCoverageStatement.coverageDepth) + '\t' +
+            sanitize(tippCoverageStatement.coverageNote) + '\t' +
+            sanitize(tipp.title.getCurrentPublisher()?.name) + '\t' +
+            sanitize(tipp.title.getPrecedingTitleId()) + '\t' +
+            (tipp.title.hasProperty('dateFirstInPrint') ? sanitize(tipp.title.dateFirstInPrint) : '') + '\t' +
+            (tipp.title.hasProperty('dateFirstOnline') ? sanitize(tipp.title.dateFirstOnline) : '') + '\t' +
+            (tipp.title.hasProperty('volumeNumber') ? sanitize(tipp.title.volumeNumber) : '') + '\t' +
+            (tipp.title.hasProperty('editionStatement') ? sanitize(tipp.title.editionStatement) : '') + '\t' +
+            (tipp.title.hasProperty('firstEditor') ? sanitize(tipp.title.firstEditor) : '') + '\t' +
+            '\t' +  // parent_publication_title_id
+            sanitize(tipp.title?.medium?.value) + '\t' +  // publication_type
+            sanitize(tipp.paymentType?.value) + '\t' +  // access_type
+            sanitize(tipp.title.getIdentifierValue('ZDB')) +
+            '\n')
+  }
+
+
```

Comment:
Both export functions (`packageKBartExport` as well as `packageTSVExport`) should be changed to utilize the PackageService methods like in `PackagesController.kbart()` & `PackagesController.packageTSVExport`.